cmd/kube-scheduler/app/options/options.go

```diff
@@ -186,15 +184,20 @@ func (o *Options) ApplyTo(c *schedulerappconfig.Config) error {
 			return err
 		}
 
-		// use the loaded config file only, with the exception of --address and --port. This means that
-		// none of the deprecated flags in o.Deprecated are taken into consideration. This is the old
-		// behaviour of the flags we have to keep.
+		// use the loaded config file only, with the exception of --address and --port.
 		c.ComponentConfig = *cfg
 
+		// if the user has set CC profiles and is trying to use a Policy config, error out
```

Comment:
You could call ApplyAlgorithmSourceTo before this and check that .AlgorithmSource.Policy is empty

Reply:
done

also note, that we can't check if `algorithmSource.Provider` is empty because when any config is provided the default algorithmProvider value is [set by conversion to the internal type](https://github.com/kubernetes/kubernetes/blob/ec36ff40016e1ea612ae364b3b550cd31dd04651/pkg/scheduler/apis/config/v1beta1/conversion.go#L53). Not sure if this matters, but I was checking against that before too